tests/queries/0_stateless/02812_pointwise_array_operations.sql

```diff
@@ -0,0 +1,8 @@
+SELECT (materialize([1,1]) + materialize([1,4]));
```

Comment:
Could you please extend the tests with:
1. problematic queries mentioned in the comments (https://github.com/ClickHouse/ClickHouse/pull/52625#discussion_r1282929807 https://github.com/ClickHouse/ClickHouse/pull/52625#discussion_r1282929807)
2. Query that the fuzzer found
3. Queries where on of the arrays is const and the other isn't
4. queries that triggers each new exception at least once
5. any query that you found interesting during debugging

Reply:
Done!